modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/object-view/ModalAddColumnsObjectCustomView.tsx

```diff
@@ -0,0 +1,262 @@
+		setFilteredItems(
+			filteredItems.map((field) => {
+				return {
+					...field,
+					checked: setSelection(checked),
+				};
+			})
+		);
+	};
+
+	const handleFieldChecked = (name: String) => {
+		const newfiltredItems = filteredItems.map((field) => {
+			if (field.name === name) {
+				return {
+					...field,
+					checked: !field.checked,
+				};
+			}
+
+			return field;
+		});
+
+		setFilteredItems(newfiltredItems);
+	};
+
+	const onSubmit = (event: FormEvent) => {
+		event.preventDefault();
+
+		dispatch({
+			payload: {
+				filteredItems,
+			},
+			type: TYPES.ADD_OBJECT_VIEW_COLUMN,
+		});
+
+		onClose();
+	};
+
+	return (
+		<>
+			<ClayModal
+				className="object-web__modal-add-columns"
+				observer={observer}
+			>
+				<ClayForm onSubmit={(event) => onSubmit(event)}>
+					<ClayModal.Header>
+						{Liferay.Language.get('add-columns')}
+					</ClayModal.Header>
+
+					<ClayModal.Body>
+						<div className="object-web__modal-add-columns--selection-title">
+							{Liferay.Language.get('select-the-columns')}
+						</div>
+
+						<ClayManagementToolbar>
+							<ClayManagementToolbar.ItemList>
+								<ClayManagementToolbar.Item>
+									<ClayCheckbox
+										checked={fieldsChecked}
+										indeterminate={
+											!allFieldsChecked && fieldsChecked
+										}
+										onChange={({target}) =>
+											handleAllFieldsChecked(
+												target.checked
+											)
+										}
+									/>
+								</ClayManagementToolbar.Item>
+							</ClayManagementToolbar.ItemList>
+
+							<ClayManagementToolbar.Search>
+								<ClayInput.Group>
+									<ClayInput.GroupItem>
+										<ClayInput
+											aria-label="Search"
+											className="form-control input-group-inset input-group-inset-after"
+											defaultValue=""
+											onChange={({target}) =>
+												setQuery(target.value)
+											}
+											placeholder={Liferay.Language.get(
+												'search'
+											)}
+											type="text"
+										/>
+
+										<ClayInput.GroupInsetItem
+											after
+											tag="span"
+										>
+											<ClayButtonWithIcon
+												className="navbar-breakpoint-d-none"
+												displayType="unstyled"
+												onClick={() => {}}
+												symbol="times"
+											/>
+
+											<ClayButtonWithIcon
+												displayType="unstyled"
+												symbol="search"
+												type="submit"
+											/>
+										</ClayInput.GroupInsetItem>
+									</ClayInput.GroupItem>
+								</ClayInput.Group>
+							</ClayManagementToolbar.Search>
+						</ClayManagementToolbar>
+
+						<ClayList className="object-web__modal-add-columns--list">
+							{filteredItems?.map((field, index) => (
+								<ClayList.Item
+									className={field?.filtered ? '' : 'hide'}
+									flex
+									key={`list-item-${index}`}
+								>
+									<ClayCheckbox
+
+										// @ts-ignore
+
+										checked={field.checked}
```

Comment:
Sorry @miltonmc, I put this `@ts-ignore` to deal with it later, but I forgot, the problem is because the checked attribute from `ClayCheckbox` only accepts the `boolean` with type, but in my field types, the checked was setting with `boolean | undefined`, I fixed this by removing `undefined` possibility from my field type.